src/RestaurantList.tsx

```diff
@@ -0,0 +1,85 @@
+import React from 'react';
+import RestaurantItem from './RestaurantItem.tsx';
+import { FilterOption, Restaurant } from './util/type.js';
+
+type StateType = {
+  restaurantList: Omit<Restaurant, 'link'>[];
+};
+
+type RestaurantListProps = {
+  filterOptions: FilterOption;
+  onToggleDrawer: (id?: number) => void;
+};
+
+class RestaurantList extends React.Component<RestaurantListProps, StateType> {
+  constructor(props) {
+    super(props);
+    this.state = {
+      restaurantList: [],
+    };
+  }
```

Comment:
별다른 연산 없이 default value를 할당하는 상황이라면 생성자 함수 없이 상단 필드에 선언하면 좋습니다🤗

```suggestion
  state = {
    restaurantList: [],
  };
```

Reply:
클래스 컴포넌트에서 state를 무조건 constructor의 this.state에서만 선언해야하는줄 알았는데 잘못 알고 있었네요 😓